torch/distributed/_spmd/comm_tensor.py

```diff
@@ -85,26 +95,29 @@ class CommTensor(torch.Tensor):
         "reduce_scatter_",
         "scatter_",
     ]
-    _after_comm: bool
+
     _tensor: torch.Tensor
-    _work: torch.distributed._Work
+    _work: Optional[torch.distributed._Work]
 
     @staticmethod
     def __new__(cls, tensor: torch.Tensor):
-        r = torch.Tensor._make_subclass(
-            cls,
-            # avoid nested CommTensor Wrapping
-            tensor._tensor if isinstance(tensor, CommTensor) else tensor,
-            require_grad=tensor.requires_grad,
-        )
+        t = tensor._tensor if isinstance(tensor, CommTensor) else tensor
+        if _get_tracer(t) is None:
+            # noop for eager mode
+            return tensor
```

Comment:
what would be the eager mode looks like then? I think for device mesh collective, we would always wrap the input tensor be CommTensor, so for "eager mode", we just return the same tensor, and user of device mesh could need to manually call `work.wait()` to sync the stream and get the results?

Reply:
> user of device mesh could need to manually call work.wait() to sync the stream and get the results?

Yep, in eager mode, users/DT will have to explicitly call wait. I assume it is already so today, otherwise it won't be semantically correct? LMK if DT relies on CommTensor to correctly wait for communication. In that case, I will revert this change and only leave clean up code.